### PR TITLE
Adding Multiple Initializations

### PR DESCRIPTION
**Pull request by jaceybronte**
This PR adds multiple random initializations for each VAE to compare overall stability and results. It also makes a few other small adjustments so that this works correctly.

### CODE REVIEW

File: 2.train-VAE/utils/betatcvae.py
Comment (gwaybio) [inline 2.train-VAE/utils/betatcvae.py:319]: I'm not sure how computationally intensive the preceding calculations are, but you could consider checking to see if the path is valid earlier. It would stink to perform all the computations and then have this function fail because of an invalid path!

Also, just to confirm, it looks like if the user provides a path argument, then the function will still output `latent_df`. Is this intended?

File: 2.train-VAE/utils/betavae.py
Comment (gwaybio) [inline 2.train-VAE/utils/betavae.py:219]: same comment here as above

File: 4.gene-expression-signatures/scripts/1.optimize_models_latent_dimensions.py
Comment (gwaybio) [inline 4.gene-expression-signatures/scripts/1.optimize_models_latent_dimensions.py:8]: It seems like you removed a space by accident?

File: 4.gene-expression-signatures/scripts/1.optimize_models_latent_dimensions.py
Comment (gwaybio) [inline 4.gene-expression-signatures/scripts/1.optimize_models_latent_dimensions.py:76]: FWIW, i recommend switching away from `os` to use `pathlib` library for all pathing related needs

File: 4.gene-expression-signatures/scripts/1.optimize_models_latent_dimensions.py
Comment (gwaybio) [inline 4.gene-expression-signatures/scripts/1.optimize_models_latent_dimensions.py:203]: This code looks like it would benefit from an autoformatter

File: 4.gene-expression-signatures/scripts/1.optimize_models_latent_dimensions.py
Comment (gwaybio) [inline 4.gene-expression-signatures/scripts/1.optimize_models_latent_dimensions.py:227]: nice! Great to save the seed as well :)

File: 4.gene-expression-signatures/scripts/1.optimize_models_latent_dimensions.py
Comment (gwaybio) [inline 4.gene-expression-signatures/scripts/1.optimize_models_latent_dimensions.py:62]: Are these functions useful enough to save in a separate utils file?

File: 4.gene-expression-signatures/scripts/2.run_gsea.py
Comment (gwaybio) [inline 4.gene-expression-signatures/scripts/2.run_gsea.py:95]: in the future, (note that I'm saying this without looking at the `weights()` function), I bet it would be better to write a single function (for example, `output_weights_matrix()`) that has an argument `model_name` rather than writing three separate functions. My intuition is that a lot of the functionality inside these three functions is redundant, and, that also you would avoid the if/else logic here if you could simply pass `model_name=model_name` in the single function.

File: 4.gene-expression-signatures/scripts/2.run_gsea.py
Comment (gwaybio) [inline 4.gene-expression-signatures/scripts/2.run_gsea.py:192]: I like these continue statements in principle, but it could be dangerous, especially if you're underlying data changes. You do have a print statement above to catch these sorts of potential errors, so they won't be completely silent. You could, alternatively, consider adding a "force" argument that will rewrite if you explicitly want it too

File: 4.gene-expression-signatures/scripts/2.run_gsea.py
Comment (gwaybio) [inline 4.gene-expression-signatures/scripts/2.run_gsea.py:219]: what does "filtered" mean in this context? Consider increasing clarity

**Review by gwaybio — APPROVED** [review]
LGTM! Mostly style, commenting, and future software improvement-based comments. Feel free to merge when you are happy.

File: 2.train-VAE/utils/betatcvae.py
Comment (jaceybronte) [reply 2.train-VAE/utils/betatcvae.py:319]: Yes, that is intentional!

File: 4.gene-expression-signatures/scripts/1.optimize_models_latent_dimensions.py
Comment (jaceybronte) [reply 4.gene-expression-signatures/scripts/1.optimize_models_latent_dimensions.py:76]: Will do!

File: 4.gene-expression-signatures/scripts/2.run_gsea.py
Comment (jaceybronte) [reply 4.gene-expression-signatures/scripts/2.run_gsea.py:95]: This makes sense yes

**Comment by gwaybio** [comment]
Checking in here on Way Lab procedures @jaceybronte 

I see that you merged this without any additional commits? Was this intentional? 

Typically, the pull request workflow is to respond to comments and make any changes here as commits, prior to merging. The "approve" ✅ means that merging without any commits is ok (you only need one approval to merge), but also the expectation is that you, as the code owner, has the strictest threshold for quality. And, therefore, that any comment that would improve your code without too much expense to productivity, would occur as a commit.

**Comment by jaceybronte** [comment]
@gwaybio 
Oh shoot no this was not intentional! The commit for those comments got in with another pull request - I will fix that now

**Comment by gwaybio** [comment]
No worries! Yes, GitHub can be tricky sometimes. Please file a new PR and reference this one. It should be a quick review since this one was already reviewed.

**Pull request merged**
